community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -459,8 +460,17 @@
     public static final Setting<Integer> label_block_size = setting("unsupported.dbms.block_size.labels", INTEGER,
             "0", min( 0 ) );
 
-    @Description("An identifier that uniquely identifies this graph database instance within this JVM. " +
-            "Defaults to an auto-generated number depending on how many instance are started in this JVM.")
+    @Description( "Specifies the size of id batches local to each transaction when committing. " +
+            "Committing a transaction which contains changes most often results in new data records being created. " +
+            "For each record a new id needs to be generated from an id generator. " +
+            "It's more efficient to allocate a batch of ids from the contended id generator, which the transaction " +
+            "holds and generates ids from while creating these new records. " +
+            "This setting specifies how big those batches are. " +
+            "Remaining ids are freed back to id generator on clean shutdown." )
+    @Internal
+    public static final Setting<Integer> record_id_batch_size = setting( "unsupported.dbms.record_id_batch_size", INTEGER,
+            "20", range( 1, 1_000 ) );
```

Comment:
20 is some empirically chosen value... it roughly amounts to 1/20 of id contention, at least something in that area. Too big will cause too many ids in flight and lost on crash. Too few will not reduce id contention enough and the management around the id batches will be overkill. Also the 1000 limit, I'm guess we should, if anything, limit it to something less because I'm thinking about id batches size is clustering environments... which I believe is in those sizes and these tx batch sizes should not be as mig as those.

Reply:
yep it should not be less than 20 for sure i was thinking about making it bigger by default and put limitations to something like 100_000 or so, for cases when your workload are batches of different sorts, even if you have a single machine or you do not really care about lost ids

Reply:
You may be correct about the upper limit. Although I'm fairly certain that the contention point in the majority of cases (mind that in most cases this id generator contention point isn't among the top points at all) a batch size of 10-20 will reduce it such that it will not be observable in any profiling. We've decided to have batch size 1 for 3.2 as default, but I'd opt for a smallish size in 3.3 and we'll see how it works from there.